src/Culling/ray.ts

```diff
@@ -564,7 +564,8 @@ export class Ray {
         nearScreenSource.x = (sourceX / viewportWidth) * 2 - 1;
         nearScreenSource.y = -((sourceY / viewportHeight) * 2 - 1);
         nearScreenSource.z = -1.0;

```

Comment:
Not related to the PR but there's a problem in WebGPU (and Native I assume) where the Z coord in NDC has a `0..1` range and not `-1..1`. So `-1` should be replaced by `1` in this case.

However, there's currently no way in this function to know that the engine has a half Z range NDC (`engine.isNDCHalfZRange`)...

Reply:
Should we create an issue for this?

Reply:
Would probably be a good idea.